### PR TITLE
feat: Recipe detail

### PR DESCRIPTION
**Pull request by taher-software**
In this branch, I've created the following changes:
If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
If the user is the owner of the recipe, should lead to the form that allows user to add new food.

### CODE REVIEW

File: app/views/recipes/show.html.erb [hunk 1]
```diff
@@ -1,10 +1,51 @@
...
+<div class="recip-general-info">
+  <div class="recip-process">
+    <p>Preparation time: <%= @recipe.preparation_time %> </p>
+    <p>Cooking time: <%= @recipe.cooking_time %> </p>
+    <p> Steps go here... </p>
+  </div>
+  <div class="public-state">
+    <label for="public"> Public</label>
+    <input type="range" name="public"  min=0 max=1 value=<%= @recipe.public %> disabled>
+  </div>
```
Comment (ShahierNashaat) [inline app/views/recipes/show.html.erb:16]: I can't change the public state here.

File: app/views/recipes/show.html.erb [hunk 2]
```diff
@@ -1,10 +1,51 @@
...
 </div>
+<div class="generating-btn" >
+  <%= button_to "Generate shopping list","/general_shopping_list" ,class:"btn btn-secondary", method: :get %>
+  <%= button_to  "Add ingredient",new_user_food_path(user_id: current_user.id), class:"btn btn-secondary", method: :get %>
```
Comment (ShahierNashaat) [inline app/views/recipes/show.html.erb:20]: Can't add a new ingredient to the recipe I just can add new food
The ingredient should be a food from the created foods before

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Please fix linter errors

File: app/views/recipes/show.html.erb (hunk 1 above)
Comment (taher-software) [reply app/views/recipes/show.html.erb:16]: The view display only information of the recipe , don't allow user to change anything

File: app/views/recipes/show.html.erb (hunk 2 above)
Comment (taher-software) [reply app/views/recipes/show.html.erb:20]: The ingredient should be a food from the created foods before !!!!
req

Comment (ShahierNashaat) [reply app/views/recipes/show.html.erb:20]: Ok but I can't see the ingredients after adding the food

Comment (taher-software) [reply app/views/recipes/show.html.erb:20]: they only require to lead the user to the form that allow to add new food. 
req_spec

**Review by ShahierNashaat — APPROVED** [review]

**Pull request merged**